pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -373,6 +391,26 @@ public void healthcheck(@Suspended AsyncResponse asyncResponse) throws Exception
             });
     }
 
+    @GET
+    @Path("/health")
+    @ApiOperation(value = "Run a healthcheck against the broker")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Everything is OK"),
+        @ApiResponse(code = 403, message = "Don't have admin permission"),
+        @ApiResponse(code = 404, message = "Cluster doesn't exist"),
+        @ApiResponse(code = 500, message = "Internal server error")})
+    @ApiParam(value = "Topic Version")
+    public void healthcheck(@Suspended AsyncResponse asyncResponse,
+                            @QueryParam("topicversion") TopicVersion topicVersion) throws Exception {
+        if (Objects.isNull(topicVersion) || topicVersion == TopicVersion.V1) {
+            healthCheck(asyncResponse, TopicVersion.V1);
+        } else if (topicVersion == TopicVersion.V2) {
```

Comment:
we don't need any extra conditional statements anymore because protected-healthcheck method already handles topic-version type. we should remove this method and make protected-healthcheck method main public-method now because it handles all scenarios.

Reply:
it doesn't handle null , this method is needed to handle that.

Reply:
it already handles null if you derive namespace
```
String namespace = (topicVersion == TopicVersion.V2) ? NamespaceService.getHeartbeatNamespaceV2( pulsar().getAdvertisedAddress(), pulsar().getConfiguration()) : NamespaceService.getHeartbeatNamespace( pulsar().getAdvertisedAddress(), pulsar().getConfiguration());
```

Reply:
ok

Reply:
changed it